src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/PageLevelErrors.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.ui.stepsdesign;
+
+import io.cdap.cdap.ui.utils.Constants;
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.WaitHelper;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+import org.openqa.selenium.WebElement;
+
+public class PageLevelErrors {
+  private final String fakeNamespace = "fakeNamespace";
+  private final String pageNotFound = "page-404-error-msg";
+  private final String pageNotFoundDefaultMessage = "page-404-default-msg";
+
+  @When("Open fake namespace home page")
+  public void openFakeNamespaceHomePage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace);
+    WaitHelper.waitForPageToLoad();
+  }
+
+  @Then("Verify Page NotFound Error Message")
+  public void verifyPageNotFoundErrorMessage() {
+    WebElement notFoundSelector = Helper.locateElementByCssSelector(
+      Helper.getCssSelectorByDataTestId(pageNotFound));
+
+    Assert.assertTrue(notFoundSelector.isEnabled());
+  }
+
+  @When("Open fake namespace studio home page")
+  public void openFakeNamespaceStudioHomePage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace + "/studio");
+    WaitHelper.waitForPageToLoad();
+  }
+
+  @When("Open fake namespace pipeline list page")
+  public void openFakeNamespacePipelineListPage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace + "/pipelines");
+    WaitHelper.waitForPageToLoad();
+  }
+
+  @When("Open fake namespace pipeline detail page")
+  public void openFakeNamespacePipelineDetailPage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace + "/view/pipelineName");
+    WaitHelper.waitForPageToLoad();
+  }
+
+  @When("Open fake namespace pipeline drafts page")
+  public void openFakeNamespacePipelineDraftsPage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace + "/pipelines/drafts");
+    WaitHelper.waitForPageToLoad();
+  }
+
+  @When("Open fake namespace Wrangler page")
+  public void openFakeNamespaceWranglerPage() {
+    SeleniumDriver.openPage(Constants.BASE_URL + "/cdap/ns/" + fakeNamespace + "/wrangler");
```

Comment:
Good to define a constant within this class for **"/cdap/ns" + fakeNamespace** and consume as it is used at many places in the class.